interop/xds/custom_lb.go

```diff
@@ -0,0 +1,161 @@
+/*
+ *
+ * Copyright 2023 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package xds contains a custom load balancing policy implementation for
+// interop tests.
+package xds
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"google.golang.org/grpc/balancer"
+	"google.golang.org/grpc/balancer/roundrobin"
+	"google.golang.org/grpc/internal/pretty"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+func init() {
+	balancer.Register(bb{})
+}
+
+const name = "test.RpcBehaviorLoadBalancer"
+
+type bb struct{}
```

Comment:
This should have a better name or be moved into a package that is only ever intended to hold this LB policy.

I'm fine with either one.  Either:

1. `mkdir interop/xds/rpcbehaviorbalancer # yuck` + `mv interop/xds/*.go interop/xds/rpcbehaviorbalancer`
2. `sed 's/bb/rpcBehaviorBB/g` or something

..............

...And now that I've reviewed the rest of the file, I think (2) makes the most sense.

Please update the docstring on the package in that case.  Just "various xds interop helpers" or something.  Or move this into the client if you want?

Reply:
Switched to rpcBehaviorBB. Kept it here as per offline discussion.